cassandra/src/main/scala/akka/stream/alpakka/cassandra/impl/CassandraSourceStage.scala

```diff
@@ -36,18 +36,22 @@ import scala.util.{Failure, Success, Try}
       }(exec)
     }
 
+    private[this] def fetch(rs: ResultSet): Unit = {
+      isFetching = true
+      // fetch next page
+      val gFut = rs.fetchMoreResults()
+      val exec = materializer.executionContext
+      GuavaFutures.invokeTryCallback(gFut, exec)(futFetchedCallback)
```

Comment:
should this be using an async call back?

Reply:
It looks OK to me, it is only called within synchronous code. 
`tryPushAfterFetch` is called via the async handler `futFetchedCallback` when the future returns.

Reply:
Yes, I just refactored it out into a method to not duplicate code.